src/sentry/reprocessing2.py

```diff
@@ -367,14 +367,16 @@ def build_event_key(primary_hash):
 
                 from sentry import eventstream
 
+                assert primary_hash is not None
+
                 # In the worst case scenario, a group will have a 1:1 mapping of primary hashes to
                 # events, which means 1 insert per event.
                 # The overall performance of this will be marginally better than the unbatched version
                 # if a group has a lot of old primary hashes.
                 eventstream.tombstone_events_unsafe(
                     project_id,
                     event_ids,
-                    old_primary_hash=old_primary_hash,
+                    old_primary_hash=primary_hash,
```

Comment:
Do I understand correctly that this bug was just a mixup of variable names? `event_ids` and `primary_hash` belong together, while `old_primary_hash` is just the hash of the event that happened to push the current batch of events over the batch size, right?

I wonder if it would make sense to move the flush code to a separate function (e.g. `flush_current_batch(project_id, group_id)`), to make further mixups impossible.

Reply:
Yeah I'll restructure the code